client/src/components/Sharing/Sharing.vue

```diff
@@ -0,0 +1,469 @@
+            return this.item.published ? "accessible via link and published" : "accessible via link";
+        },
+        itemRoot() {
+            const port = window.location.port ? `:${window.location.port}` : "";
+            return `${window.location.protocol}//${window.location.hostname}${port}${getAppRoot()}`;
+        },
+        itemUrl() {
+            return `${this.itemRoot}${this.item.username_and_slug}`;
+        },
+        itemSlugParts() {
+            const str = this.item.username_and_slug;
+            const index = str.lastIndexOf("/");
+            return [str.substring(0, index + 1), str.substring(index + 1)];
+        },
+        itemUrlParts() {
+            const str = this.itemUrl;
+            const index = str.lastIndexOf("/");
+            return [str.substring(0, index + 1), str.substring(index + 1)];
+        },
+        published_url() {
+            return `${getAppRoot()}${this.pluralNameLower}/list_published`;
+        },
+        slugUrl() {
+            return `${getAppRoot()}api/${this.pluralNameLower}/${this.id}/slug`;
+        },
+    },
+    created: function () {
+        this.getModel();
+    },
+    methods: {
+        addError(newError) {
+            // temporary turning Set into Array, until we update till Vue 3.0, that supports Set reactivity
+            this.errors = Array.from(new Set(this.errors).add(newError));
+        },
+        onCopy() {
+            copy(this.itemUrl);
+            this.tooltipClipboard = "Copied!";
+        },
+        onCopyOut() {
+            this.tooltipClipboard = "Copy URL";
+        },
+        onEdit() {
+            this.showUrl = false;
+        },
+        assignItem(newItem) {
+            if (newItem.errors) this.errors = newItem.errors;
+            this.item = newItem;
+            if (!this.item.extra || newItem.errors.length > 0) {
+                this.item.extra = defaultExtra();
+            }
+
+            this.ready = true;
+        },
+        onChange(newSlug) {
+            this.showUrl = true;
+            const requestUrl = `${this.slugUrl}`;
+            axios
+                .put(requestUrl, {
+                    new_slug: newSlug,
+                })
+                .then((response) => {
+                    this.errMsg = null;
+                    this.assignItem(response.data);
+                    this.item.username_and_slug = `${this.itemSlugParts[0]}${newSlug}`;
+                })
+                .catch((error) => this.addError(error.response.data.err_msg));
+        },
+        onImportable(importable) {
+            if (importable) {
+                this.setSharing(this.actions.enable_link_access);
+            } else {
+                this.item.published = false;
+                this.setSharing(this.actions.disable_link_access);
+            }
+        },
+        onPublish(published) {
+            if (published) {
+                this.item.importable = true;
+                this.setSharing(this.actions.publish);
```

Comment:
Shouldn't this be `this.setSharing(undefined, this.actions.publish);` (and everywhere else where you pass a single `share_option` argument)? My JS knowledge might be failing me here.

Reply:
Sorry, I didn't get you.
`setSharing(action, user_id, share_option) `
if you call `this.setSharing(undefined, this.actions.publish);`, then `action` would be `undefined` and `user_id` is `this.actions.publish`.

Reply:
Ah, I think I missed that the first argument is `action`. Yeah, that makes sense.